### PR TITLE
Update for lock.go

### PR DESCRIPTION
**Pull request by diaevd**
Переписанный (упрощенный) lock.go с одним интерфейсом KeyLock.

kl := NewKeyLock
kl.Block("key")
kl.Unblock("key")

и простенький тест, как сделать нормальный тест пока не понял.

### CODE REVIEW

File: lock.go [hunk 1]
```diff
@@ -1,71 +1,51 @@
...
+	mutex        map[string]*sync.Mutex
 }
 
-// интерфейс блокировки по строковому ключу
-type KeyLocker interface {
-	Locker
-	NewKey()
-}
-
-type lock struct {
```
Comment (mtfelian) [inline lock.go:19]: Don't delete it, it is used in many packages

File: lock_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+			kl.Unblock(in.key)
+			fmt.Printf("%v After  Unblock('%v'): %v\n", time.Since(start), in.key, in.sleep)
+		}(p)
+	}
+
+	// Использовать sync.WaitGroup чтобы дождаться окончания работы
+	// всех go-рутин не получится в данном случае.
+	// можно только по таймеру, так как сам WaitGroup, зараза,
+	// лочится изнутри свои mux-ом и мы получаем deadlock.
+	// поэтому Sleep и никак иначе (есть другие варианты?)
```
Comment (mtfelian) [inline lock_test.go:41]: Don't understand, how is it WG mutex interferes with local mutex?

**Review by mtfelian — CHANGES_REQUESTED** [review]
Backward compatibility should be retained

File: lock.go [hunk 3]
```diff
@@ -34,38 +32,57 @@ func (obj *lock) Unblock() {
...
-	return &LockKey{make(map[string]*sync.Mutex), make(map[string]*sync.Once)}
+
+	return &LockKey{
+		mutex: make(map[string]*sync.Mutex),
+	}
 }
 
-func (obj *LockKey) newKey(key string) {
-	obj.mutex[key] = &sync.Mutex{}
-	obj.once[key] = &sync.Once{}
```
Comment (mtfelian) [inline lock.go:48]: Why no more Once here ?

Comment (diaevd) [reply lock.go:48]: this no need anywere

**Comment by diaevd** [comment]
Fix deadlock

File: lock_test.go (hunk 2 above)
Comment (diaevd) [reply lock_test.go:41]: This is was my bug, it realy did deadlock

**Pull request merged**